test/Microsoft.AspNetCore.Mvc.TagHelpers.Test/CacheTagHelperTest.cs

```diff
@@ -648,6 +648,67 @@ public void UpdateCacheEntryOptions_SetsCachePreservationPriority()
             Assert.Equal(2, calls);
         }
 
+        [Fact]
+        public async Task ProcessAsync_ExceptionInProcessing_DoNotThrowInSubsequentRequests()
+        {
+            // Arrange
+            var id = "unique-id";
+            var childContent = "some-content";
+            var cache = new MemoryCache(new MemoryCacheOptions());
+
+            var tagHelperContext1 = GetTagHelperContext(id);
+            var tagHelperContext2 = GetTagHelperContext(id);
+
+            var tagHelperOutput1 = new TagHelperOutput(
```

Comment:
You are testing two different helpers/outputs using the same key on purpose. What I'd like to see is use the same helper/output with an invocation count that would throw on the first two and then return on the third. This would ensure the same "faulted" task still works and also is not cached.